flashflood/identifiers.py

```diff
@@ -0,0 +1,98 @@
+from uuid import uuid4
+from enum import Enum
+from functools import lru_cache
+
+from flashflood.util import datetime_from_timestamp, timestamp_now
+
+
+TOMBSTONE_SUFFIX = ".dead"
+
+
+class JournalUpdateAction(Enum):
+    UPDATE = "update"
+    DELETE = "delete"
+
+
+class JournalID(str):
```

Comment:
Should add a docstring here that briefly explains what a journal ID is/what it is used for/what form it takes.

```
A Journal ID is a string that identifies journal events by their start/end date, version, and blobs affected. (...)
```

Maybe this is clear to you from the name and from reading the code, but it's a little confusing to me because there are Journal updates, actions, events, IDs, update IDs, etc., some of which are represented by classes, and it's hard to parse out how these are related or unrelated.